src/App.tsx

```diff
@@ -17,18 +56,31 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filterSelection={filterSelection}
+                setFilterSelection={setFilterSelection}
+                query={query}
+                setQuery={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length === 0 && <Loader />}
```

Comment:
will be great to handle it with separate variable, that will be responsible for the data loading
for now if you'll receive an empty array from server, user will see infinite loader

Reply:
(not important)